assembl/models/generic.py

```diff
@@ -549,6 +555,52 @@ def widget_ideas(self):
         from .idea import Idea
         return [Idea.uri_generic(wil.idea_id) for wil in self.widget_idea_links]
 
+    def nlp_keywords(self, display_lang=None, limit=20):
+        from .nlp import PostKeywordAnalysis, Tag
+        from .langstrings import LangStringEntry, Locale
+
+        sq = self.db.query(
+            PostKeywordAnalysis.score.label('score'), Tag.id.label('id')
+        ).filter_by(
+            category=None
+        ).join(Tag).filter(
+            PostKeywordAnalysis.post_id == self.id
+        ).order_by(
+            PostKeywordAnalysis.score.desc())
+        if limit:
+            sq = sq.limit(20)
```

Comment:
ah, interesting. Was that ever an issue?
I'm adjusting, but rather with an assert. Otherwise, a caller giving a bad limit will fail silently, and not know that it did anything wrong.